API: .dt.components attribute on timedelta series does not preserve the index

```
In [70]: tds = pd.Series(pd.timedelta_range('1 days', periods=5, freq='2 h'), index=list('ABCDE'))

In [71]: tds
Out[71]:
A   1 days 00:00:00
B   1 days 02:00:00
C   1 days 04:00:00
D   1 days 06:00:00
E   1 days 08:00:00
dtype: timedelta64[ns]

In [72]: tds.dt.hours
Out[72]:
A    0
B    2
C    4
D    6
E    8
dtype: int64

In [73]: tds.dt.components.hours
Out[73]:
0    0
1    2
2    4
3    6
4    8
Name: hours, dtype: int64
```
So the direct attributes via `.dt` like `.dt.hours` above does preserve the index, but the `components` not. 

@jreback I don't know if this was done on purpose or is an oversight, I can't remember, and I looked briefly to the long discussion in #8184, but didn't find something directly.

--- a/doc/source/whatsnew/v0.16.0.txt
+++ b/doc/source/whatsnew/v0.16.0.txt
@@ -86,7 +86,7 @@ Bug Fixes
 - Fixed issue using `read_csv` on s3 with Python 3.
 - Fixed compatibility issue in ``DatetimeIndex`` affecting architectures where ``numpy.int_`` defaults to ``numpy.int32`` (:issue:`8943`)
 - Bug in Panel indexing with an object-like (:issue:`9140`)
-
+- Bug in the returned ``Series.dt.components`` index was reset to the default index (:issue:`9247`)
 
 
 

--- a/pandas/tests/test_series.py
+++ b/pandas/tests/test_series.py
@@ -142,7 +142,7 @@ class CheckNameIntegration(object):
             tm.assert_series_equal(result, expected)
 
         # timedeltaindex
-        for s in [Series(timedelta_range('1 day',periods=5)),
+        for s in [Series(timedelta_range('1 day',periods=5),index=list('abcde')),
                   Series(timedelta_range('1 day 01:23:45',periods=5,freq='s')),
                   Series(timedelta_range('2 days 01:23:45.012345',periods=5,freq='ms'))]:
             for prop in ok_for_td:

--- a/pandas/tseries/common.py
+++ b/pandas/tseries/common.py
@@ -139,7 +139,7 @@ class TimedeltaProperties(Properties):
 
     @property
     def components(self):
-        return self.values.components
+        return self.values.components.set_index(self.index)
 
 TimedeltaProperties._add_delegate_accessors(delegate=TimedeltaIndex,
                                             accessors=TimedeltaIndex._datetimelike_ops,

